content/04.workflows-and-software.md

```diff
@@ -42,8 +42,8 @@ These tools allow users to take advantage of the benefits of workflow software w
 If your use case extends beyond these tools, there are several scriptable workflow systems that offer comparable benefits for data intensive biology.
 Each has it own strengths, meaning each software will meet an individuals computing goals differently (see **Table @tbl:workflows**).
 Our lab has adopted Snakemake, in part due to its similarity and integration with Python, its flexibility for building and testing new analyses in different languages, and its intuitive integration with software management tools (described below)[@doi:10.1093/bioinformatics/bts480].
-Software like Nextflow and Common Workflow Language require slightly more infrastructure, but in return, scale much better to pipelines with hundreds of thousands of steps and support containerization more rigidly, making them ideal for production-level pipelines [@doi:10.1038/nbt.3820; @doi:10.6084/m9.figshare.3115156.v2].
-Language-specific workflow systems, such as ROpenSci's Drake [@doi:10.21105/joss.00550], are limited in the scope of tasks they can execute, but are powerful within their language and easier to integrate if comfortable in that language.
+Software like Nextflow and Common Workflow Language require slightly more infrastructure, but in return, scale much better to pipelines with hundreds of thousands of steps and support containerization more rigidly (CTB what are you trying to say here??), making them ideal for production-level pipelines [@doi:10.1038/nbt.3820; @doi:10.6084/m9.figshare.3115156.v2].
```

Comment:
```suggestion
Software like Nextflow and Common Workflow Language require slightly more infrastructure, but in return, scale much better to pipelines with hundreds of thousands of steps, making them ideal for production-level pipelines [@doi:10.1038/nbt.3820; @doi:10.6084/m9.figshare.3115156.v2].
```

Just trying to give comparative merits between workflow systems. But now that I'm thinking about it, the "more infrastructure" part may only apply to nextflow pipelines considered for nf-core. Is there a better way to get across the difference that snakemake builds the entire dag first, which means that scaling up can currently be frustrating?

Reply:
I was specifically flagging the "support containerization more rigidly" text.

Reply:
I don't know what that phrase means :)

Reply:
I would ignore scaling question. Sorta off topic for this paper.

Reply:
@olgabot, do you have any insight here? We want to make sure we highlight Nextflow and CWL, but other than scaling, we're not sure how to best present the differences or trade offs. I think it's true that NF and CWL are more common in production level code, but I don't think that's always true.

Reply:
Yes, I don't think that nextflow is particularly more scalable than snakemake. CWL and WDL runners tend to think more on the "let's run 100,000 jobs now" side and so they ARE (or should be) more scalable, but I don't know of citations or objective data.

Reply:
Maybe the way to proceed is to distinguish Nextflow and Snakemake as research workflow systems that we explicitly do recommend for research bioinformaticians doing bespoke research pipeline development, vs CWL and WDL which are interchange formats that I think are best built and used through tools (like Rabix/SBG, or Terra). We could run something by luiz, camille, or michael crusoe if we wanted to double-check the language.

Reply:
gave it a shot! I'll make a note to specifically ask for language check/feedback on this section when we send out to coauthors

Reply:
nice!